### PR TITLE
fix url check for jolokia()

### LINKED ISSUE #365: Fix http().jolokia() sending arbitrary POST requests
The safety check in jolokia can be circumvented and use can use it to send arbitrary POST requests.

https://github.com/zalando-zmon/zmon-worker/blob/master/zmon_worker_monitor/builtins/plugins/http.py#L258

### PR DESCRIPTION
**Pull request by vetinari**
by parsing the URL and checking the path instead of the full url

fixes #365

### CODE REVIEW

**Comment by vetinari** [comment]
:+1:

**Comment by vetinari** [comment]
:+1:

File: zmon_worker_monitor/builtins/plugins/http.py [hunk 1]
```diff
@@ -255,10 +255,14 @@ def set_read_type(x):
...
-        if (not self.url.endswith('jolokia/')) or ('?' in self.url) or ('&' in self.url):
-            raise HttpError("URL needs to end in jolokia/ and not contain ? and &", self.url)
+        parsed_url = urlparse.urlsplit(self.url)
+        if (not parsed_url.path.endswith('/jolokia/')) or ('?' in self.url) or ('&' in self.url):
+            raise HttpError("URL needs to end in /jolokia/ and not contain ? and &", self.url)
 
         map(set_read_type, read_requests)
+        for rr in read_requests:
+            if 'mbean' not in rr:
+                raise HttpError('missing "mbean" key in read request')
```
Comment (mohabusama) [inline zmon_worker_monitor/builtins/plugins/http.py]: Let's raise a ``CheckError`` here

**Comment by vetinari** [comment]
:+1:

**Comment by vetinari** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**